WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -217,10 +222,23 @@ def create
       registration_competition_events_attributes: [:id, :competition_event_id, :_destroy],
     ]
     if current_user.can_manage_competition?(competition_from_params)
-      permitted_params << :accepted_at
+      permitted_params << [
+        :accepted_at,
+        :accepted_by,
+        :deleted_at,
+        :deleted_by,
+      ]
       status = params[:registration][:status]
-      if status
-        params[:registration][:accepted_at] = (status == "a" ? Time.now : nil)
+      if status == "a"
+        params[:registration][:accepted_at] = Time.now
+        params[:registration][:accepted_by] = current_user.id
+        params[:registration][:deleted_at] = nil
+      elsif status == "d"
+        params[:registration][:deleted_at] = Time.now
+        params[:registration][:deleted_by] = current_user.id
+      else
```

Comment:
Can we be explicit about the possible values of `status` here? If we get something unrecognized, I say we raise an exception.